vulnerabilities/importers/github.py

```diff
@@ -146,189 +122,339 @@
-        asyncio.run(self.version_api.load_api(packages))
 
-    def updated_advisories(self) -> Set[Advisory]:
-        return self.batch_advisories(self.process_response())
+class InvalidVersionRange(Exception):
+    pass
 
-    def fetch(self) -> Mapping[str, List[Mapping]]:
-        headers = {"Authorization": "token " + self.gh_token}
-        api_data = {}
-        for ecosystem in self.config.ecosystems:
 
-            api_data[ecosystem] = []
-            end_cursor_exp = ""
+def process_name_namespace(ecosystem: str, name: str, namespace: str) -> str:
+    """
+    Return `namespace` and `name` by processing `ecosystem` and `pkg_name`
+    """
+    if ecosystem == "MAVEN":
+        return f"{namespace}:{name}"
 
-            while True:
-                query_json = {"query": query % (ecosystem, end_cursor_exp)}
-                resp = requests.post(self.config.endpoint, headers=headers, json=query_json).json()
-                if resp.get("message") == "Bad credentials":
-                    raise GitHubTokenError("Invalid GitHub token")
+    if ecosystem == "COMPOSER":
+        return f"{namespace}/{name}"
 
-                end_cursor = resp["data"]["securityVulnerabilities"]["pageInfo"]["endCursor"]
-                end_cursor_exp = "after: {}".format('"{}"'.format(end_cursor))
-                api_data[ecosystem].append(resp)
+    if ecosystem in ("NUGET", "PIP", "RUBYGEMS", "GO"):
+        return name
 
-                if not resp["data"]["securityVulnerabilities"]["pageInfo"]["hasNextPage"]:
-                    break
-        return api_data
-
-    def set_version_api(self, ecosystem: str) -> None:
-        versioners = {
-            "MAVEN": MavenVersionAPI,
-            "NUGET": NugetVersionAPI,
-            "COMPOSER": ComposerVersionAPI,
-            "PIP": PypiVersionAPI,
-            "RUBYGEMS": RubyVersionAPI,
-            "GO": GoproxyVersionAPI,
-        }
-        versioner = versioners.get(ecosystem)
-        if versioner:
-            self.version_api = versioner()
-            self.set_api(self.collect_packages(ecosystem))
-
-    @staticmethod
-    def process_name(ecosystem: str, pkg_name: str) -> Optional[Tuple[Optional[str], str]]:
-        if ecosystem == "MAVEN":
-            artifact_comps = pkg_name.split(":")
-            if len(artifact_comps) != 2:
-                return
-            ns, name = artifact_comps
-            return ns, name
-
-        if ecosystem == "COMPOSER":
+
+def process_response(resp: dict, ecosystem: str) -> Iterable[AdvisoryData]:
+    """
+    Yield `AdvisoryData` by taking
+    `resp` and `ecosystem` as input
+    """
+    edges = get_item(resp, "data", "securityVulnerabilities", "edges") or []
+    if not edges:
+        LOGGER.error(f"No edges found in response {resp!r}")
+    for adv in edges:
+        affected_packages = []
+        cve_ids = set()
+        node = get_item(adv, "node")
+        if not node:
+            LOGGER.error(f"No node found in {adv!r}")
```

Comment:
Node and edges feel super generic.... what could be a more explicit message that can make sense without knowing the internals of the GitHub GraphQL API?